Source/UI/Coordinators/PasswordCoordinator.swift

```diff
@@ -194,4 +229,72 @@ extension PasswordCoordinator {
         }
         self.navigationController.pushViewController(viewController, animated: true)
     }
+    private func getPasswordFromKeychain(for identifier: Identifier) -> String? {
+        if #available(iOS 11.3, *) {
+            var query = [String: Any]()
+    
+            query[kSecClass as String] = kSecClassGenericPassword
+            query[kSecReturnData as String] = kCFBooleanTrue
+            query[kSecAttrAccount as String] = identifier.normalizedString as CFString
+            query[kSecAttrLabel as String] = "com.schibsted.account.biometrics.secrets" as CFString
+            query[kSecUseOperationPrompt as String] = "Please put your fingers on that button" as CFString
+    
+            var queryResult: AnyObject?
+            let status = withUnsafeMutablePointer(to: &queryResult) {
+                SecItemCopyMatching(query as CFDictionary, UnsafeMutablePointer($0))
+            }
+    
+            if status == noErr {
+                let password = String(data: queryResult as! Data, encoding: .utf8)!
+                print("Password: \(password)")
+                return password
+            } else {
```

Comment:
This could crash the app if the cast to `Data` fails. This can be avoided with optional binding:
```
if status == noErr, let qresult = queryResult as? Data {
    return password
} else {
```

Reply:
Fixed!